docs/proposals/cluster-default-components-settings.md

```diff
@@ -0,0 +1,31 @@
+# Cluster default component settings webhook
+
+**Author**: Furkhat @furkhat
+
+**Status**: Draft proposal; prototype in progress.
+
+## Goals
+
+*short description of the topic e.g.*
+Add mutating admission webhook that reads configured configmap with default components settings and sets `.spec.componentsOverride` for new clusters.
```

Comment:
As I mentioned [here](https://github.com/kubermatic/kubermatic/pull/5756#issuecomment-730229579) I would prefer introducing a new CRD, e.g. `ComponentSettingsPreset` that includes the component settings and a label selector to match the clusters (similarly to https://github.com/kubernetes/community/blob/master/contributors/design-proposals/service-catalog/pod-preset.md). Not sure if I'm overthinking here, but this approach seems more flexible and future-proof than having a static configuration at `seed-controller-manager` level.

I imagine that KKP administrators could provide several cluster flavours, that could be selected at cluster creation time by the platform users (UI could allow choosing between the available presets in the cluster creation wizard). @zreigz @shaase-ctrl  Any thoughts on this?

Reply:
No I think we should implement the simplest approach for our current needs and don't over engineer. I can imagine `ComponentSettingsPreset` could be useful but we don't need it at the moment. And I think that could be added latter on.

With this proposal not much have changed from users perspective because this approach doesn't really change the way things work. Instead of cli flags and reconciling defaults for all the clusters all the time the proposal is to set defaults at creation instead.

Reply:
@furkhat I agree with the idea of defaulting instead of reconciling, and that we could start with a simple approach and building on top if needed, but would be better if we can do it in an incremental way and trying to avoid another breaking change later.

One aspect I don't like very much with this approach is that it brings a start time dependency on `seed-controller-manager`. If I understood it correctly, the config-map containing the settings should be there when `seed-controller-manager` starts. Practically speaking this means that there is an impact on the operator and on the helm charts too, that should be modified to deploy the cm. You could also decide to read the cm dynamically each time there is a cluster admission, but at this point, I would prefer a CRD based approach because it's more structured, and we can use the admission mechanism for validation. To keep it simple we could also decide that to start with, only one CRD instance is admitted (similarly to what is done for global Kubermatic config).

Other folks asked for a similar feature, I will try to put them into the loop to see if this approach you proposed fits their needs too.